### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by halkeye**
# Make docker-compose work again
* Mount src into the container so you can test local changes
* Fix Postgres version to the one that works with the version of PHP
* Move files around

# Clean up ratings

* Ignore ratings for non-versions
* Support ?json=1 and ?jsonp=callback so they can work on dev sites
* use full JSON encoding instead of manual creating strings

### CODE REVIEW

**Comment by halkeye** [comment]
@daniel-beck @dduportal @timja @jenkins-infra/ci-maintainers

**Comment by halkeye** [comment]
just FYI, I'm selfishly doing it because you can only access ratings.jenkins.io via javascript when you are on jenkins.io
That means neither jenkins.io repo, nor my fork can access it.

This would expose a JSONp, and a raw JSON version that would be cross-domain. It doesn't change being able to submit, that has to be locked down to jenkins.io, but it does allow reading. Plus its a little less hacky and a little more standards.

File: test/config/dbconfig.php [hunk 1]
```diff
@@ -4,4 +4,3 @@
 $dbname='rating';
 $dbserver='postgres';
 $dbport='';
-?>
```
Comment (timja) [inline test/config/dbconfig.php:7]: did you mean to delete this?

Comment (timja) [reply test/config/dbconfig.php:7]: optional apparently https://www.php.net/manual/en/language.basic-syntax.instruction-separation.php#:~:text=The%20closing%20tag%20of%20a%20PHP%20block%20at%20the%20end,headers%20to%20the%20response%20later.

**Review by timja — APPROVED** [review]

File: test/config/dbconfig.php (hunk 1 above)
Comment (halkeye) [reply test/config/dbconfig.php:7]: yea generally good practice with php so you don't have newline errors with includes.

**Comment by halkeye** [comment]
sweet, now we just need to figure out how to deploy the new file.

**Comment by timja** [comment]
There's a docker image that hasn't been built in 5 years, not sure if it's that.

The build fails because there's no credentials specified: https://github.com/jenkins-infra/rating/blob/master/Jenkinsfile

just needs the standard auth infra.with something that the other repoes have

**Comment by halkeye** [comment]
Its not deployed via k8s i know that, that project never got finished.
ratings.jenkins.io resolves to 52.23.130.110
which is an aws instance. I feel like its a manual deploy

**Comment by timja** [comment]
Probably running in docker on a VM, I’ll try ssh to it later but depends if puppet is running on it

**Comment by halkeye** [comment]
https://github.com/jenkins-infra/rating/pull/15
https://github.com/jenkins-infra/charts/pull/1650

build on the infra ...infra

**Comment by timja** [comment]
This line needs updating once a new image has been pushed: https://github.com/jenkins-infra/jenkins-infra/blob/df1a23b54bce3328f7466dc4fb4623a1173925ab/hieradata/common.yaml#L162

**Comment by halkeye** [comment]
https://github.com/jenkins-infra/jenkins-infra/pull/1976

File: src/result.php [hunk 2]
```diff
@@ -1,36 +1,54 @@
...
-  foreach ($data as $version => $info) {
-    if ($first) $first = false; else print ",\n";
-    foreach (array('good','rollback','nolike') as $key) if (!isset($info[$key])) $info[$key] = 0;
-    print json_encode($version) . ": [$info[good],$info[nolike],$info[rollback]";
-    if ($info[issues]) {
-      $issues = array_count_values($info['issues']);
-      arsort($issues);
-      foreach ($issues as $issue => $counts) {
-        print ",$issue,$counts";
-      }
```
Comment (daniel-beck) [inline src/result.php:31]: This was intentional, to sort most reported issues first in the list.

It's broken now:

> Screenshot 2021-12-28 at 09 54 00

Comment (daniel-beck) [reply src/result.php:31]: Filed https://issues.jenkins.io/browse/INFRA-3170

Comment (halkeye) [reply src/result.php:31]: https://github.com/jenkins-infra/jenkins.io/pull/4799

Comment (daniel-beck) [reply src/result.php:31]: Thanks!

**Pull request merged**